nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-multi/src/main/java/org/nuxeo/ecm/directory/multi/MultiReference.java

```diff
@@ -83,36 +89,30 @@ public void addLinks(List<String> sourceIds, String targetId) throws DirectoryEx
                 }
             }
         }
-        List<String> x = new ArrayList<String>(ids.size());
+        List<String> x = new ArrayList<>(ids.size());
         x.addAll(ids);
         return x;
     }
 
     @Override
     public List<String> getSourceIdsForTarget(final String targetId) throws DirectoryException {
-        return doCollect(new Collector() {
-            @Override
-            public List<String> collect(List<Reference> refs) throws DirectoryException {
-                List<String> sourceIds = new ArrayList<>(1);
-                for (Reference ref : refs) {
-                    sourceIds.addAll(ref.getSourceIdsForTarget(targetId));
-                }
-                return sourceIds;
+        return doCollect(refs -> {
+            List<String> sourceIds = new ArrayList<>(1);
+            for (Reference ref : refs) {
+                sourceIds.addAll(ref.getSourceIdsForTarget(targetId));
             }
+            return sourceIds;
```

Comment:
- Do you know why there's `ArrayList<>(1)` ? When I read the code it seems that we can have more than one value.
- Maybe something using stream is better here, as the jvm will create the array list depending on the element size, for instance: `refs.stream().flatMap(ref -> ref.getSourceIdsForTarget(targetId).stream()).collect(Collector.toList());`